sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -290,19 +291,19 @@ private void setBatchType(BlobBatchType batchType) {
         // Begin a new batch.
         batchOperationQueue = new ConcurrentLinkedDeque<>();
 
-        return Flux.generate(sink -> {
-            if (operations.isEmpty()) {
-                operationInfo.finalizeBatchOperations();
-                sink.complete();
-            } else {
-                BlobBatchOperation<?> batchOperation = operations.pop();
-                sink.next(batchOperation.getResponse()
-                    .subscriberContext(Context.of(BATCH_REQUEST_URL_PATH, batchOperation.getRequestUrlPath(),
-                        BATCH_OPERATION_RESPONSE, batchOperation.getBatchOperationResponse(),
-                        BATCH_OPERATION_INFO, operationInfo))
-                    .subscribe());
-            }
-        }).then(Mono.just(operationInfo));
+        List<Mono<? extends Response<?>>> batchOperationResponses = new ArrayList<>();
```

Comment:
Can you leave a comment about how this behavior is safer and resolves the issue in question so we have some context in case we ever have to investigate it later?

Reply:
Added a comment explaining the change and why it is safer and resolves the issue.